jedi/evaluate/docscrape_google.py

```diff
@@ -0,0 +1,308 @@
+    # to differentiate between the type hint and the description. This can
+    # cause an issue if the returns block only contains a description and no
+    # type hint. This will work for the majority of cases, but in the future
+    # this should be implemented using a parser.
+    retdict_list = []
+    noindent_pat = re.compile('^[^\s]')
+    for line in lines.split('\n'):
+        if noindent_pat.match(line):
+            parts = line.split(':')
+            type_part = parts[0]
+            # desc_part = ':'.join(parts[1:])
+            # retdict = {'type': type_part, 'desc': desc_part}
+            retdict = {'type': type_part}
+            retdict_list.append(retdict)
+    return retdict_list
+
+
+def parse_google_argblock(lines):
+    r"""
+    Args:
+        lines (str): the unindented lines from an Args docstring section
+
+    References:
+        # It is not clear which of these is *the* standard or if there is one
+        https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google
+        http://www.sphinx-doc.org/en/stable/ext/example_google.html#example-google
+
+    Example:
+        >>> from jedi.evaluate.docscrape_google import *  # NOQA
+        >>> # Test various ways that arglines can be written
+        >>> line_list = [
+        ...     '',
+        ...     'foo1 (int): a description',
+        ...     'foo2: a description\n    with a newline',
+        ...     'foo3 (int or str): a description',
+        ...     'foo4 (int or threading.Thread): a description',
+        ...     #
+        ...     # this is sphynx-like typing style
+        ...     'param1 (:obj:`str`, optional): ',
+        ...     'param2 (:obj:`list` of :obj:`str`):',
+        ...     #
+        ...     # the Type[type] syntax is defined by the python typeing module
+        ...     'attr1 (Optional[int]): Description of `attr1`.',
+        ...     'attr2 (List[str]): Description of `attr2`.',
+        ...     'attr3 (Dict[str, str]): Description of `attr3`.',
+        ... ]
+        >>> lines = '\n'.join(line_list)
+        >>> argdict_list = list(parse_google_argblock(lines))
+        >>> # print('argdict_list = %s' % (argdict_list),)
+        >>> # All lines except the first should be accepted
+        >>> assert len(argdict_list) == len(line_list) - 1
+    """
+    name_pat = r'(?P<name>[A-Za-z_][A-Za-z0-9_]*)'
+    type_pat = r'(?P<type>[^)]*)'
+    # Typing is optional
+    or_parts = [
+        '\(' + type_pat + '\)\s*:',
+        '\s*:'
+    ]
+    type_part = '(' + '|'.join(or_parts) + ')'
+    # Each arg hint must defined a on newline without any indentation
+    argline_pat = '^' + name_pat + r'\s*' + type_part
+
+    argdict_list = []
+    for match in re.finditer(argline_pat, lines, flags=re.M):
+        argdict = match.groupdict()
+        type_ = argdict['type']
+        if type_ is not None:
+            pass
+        argdict_list.append(argdict)
+    return argdict_list
```

Comment:
This is where each argline is parsed after it is known to belong to an Args block